### PR TITLE
API Moving index finding to a static function

### PR DESCRIPTION
**Pull request by dhensby**
For a 2.3 release

Moving this out to it's own function so other modules can use it

### CODE REVIEW

File: code/solr/Solr.php [hunk 1]
```diff
@@ -140,6 +140,38 @@ public static function get_indexes()
...
+            foreach(self::get_indexes() as $solrIndexClass) {
+                $reflection = new ReflectionClass($solrIndexClass);
+                //skip over abstract classes
+                if (!$reflection->isInstantiable()) {
+                    continue;
+                }
+                //check the indexname matches the index passed to the request
+                if (!strcasecmp(singleton($solrIndexClass)->getIndexName(), $indexName)) {
+                    //if we match, set the correct index name and move on
+                    $index = $solrIndexClass;
```
Comment (tractorcow) [inline code/solr/Solr.php]: Just return here?

File: code/solr/Solr.php [hunk 2]
```diff
@@ -140,6 +140,38 @@ public static function get_indexes()
...
+                    break;
+                }
+            }
+        }
+
+        return $index;
```
Comment (tractorcow) [inline code/solr/Solr.php]: Where's the error where an invalid $index is passed in? It shouldn't return $index if that's not a valid name.

File: code/solr/Solr.php [hunk 3]
```diff
@@ -140,6 +140,38 @@ public static function get_indexes()
...
+        $index = $indexName;
+
+        //find the index class by IndexName
+        // this is for when index names do not match the class name
+        // (this can be done by overloading getIndexName() on indexes
+        if ($indexName && !ClassInfo::exists($index)) {
+            foreach(self::get_indexes() as $solrIndexClass) {
+                $reflection = new ReflectionClass($solrIndexClass);
+                //skip over abstract classes
+                if (!$reflection->isInstantiable()) {
```
Comment (tractorcow) [inline code/solr/Solr.php]: FullTextSearch::get_indexes() already filters these.

**Comment by robbieaverill** [comment]
@dhensby would you like this to be revived again?

**Comment by robbieaverill** [comment]
Feel free to reopen if you'd like to do this in a newer release line =)

**Pull request closed without merging**